tsdb/index/tsi1/partition.go

```diff
@@ -322,9 +323,24 @@ func (p *Partition) buildSeriesSet() error {
 	return nil
 }
 
-// Wait returns once outstanding compactions have finished.
+// CurrentCompactionN returns the number of compactions currently running.
+func (p *Partition) CurrentCompactionN() int {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+	return p.currentCompactionN
+}
+
+// Wait will block until all compactions are finished.
+// Must only be called while they are disabled.
 func (p *Partition) Wait() {
-	p.wg.Wait()
+	ticker := time.NewTicker(10 * time.Millisecond)
```

Comment:
I forgot to add that we should include the changes I made to help with this. Can you cherry pick 030083e1a368725799b2a20b07d2f61d578778b5 into this PR?

Reply:
I pulled in the change but slightly tweaked it so it only has one invocation of `CurrentCompactionN()`

https://github.com/influxdata/influxdb/blob/23754e4b259548ef1295e38097fb3aea2bc9009f/tsdb/index/tsi1/partition.go#L333-L344